app/components/Article/Contents.tsx

```diff
@@ -9,6 +9,21 @@ import {createRoot} from 'react-dom/client'
 import MediaCarousel from '../MediaCarousel'
 import {Carousel} from '~/server-utils/parsing-utils'
 
+// Page-level cache for glossary terms that have been seen
+// This ensures each term is only marked once per page, regardless of how many Contents components exist
+const pageGlossaryState = {
+  seen: new Set<string>(),
+  currentPageId: null as string | null,
+}
+
+const getOrResetSeenSet = (pageid: string): Set<string> => {
+  if (pageGlossaryState.currentPageId !== pageid) {
+    pageGlossaryState.seen = new Set<string>()
+    pageGlossaryState.currentPageId = pageid
+  }
+  return pageGlossaryState.seen
+}
```

Comment:
I'm a bit skeptical this global pageGlossaryState approach though it might be okay in practice. A problem that I see is that there is an assumption that all `Contents.tsx` rendered will be for the same page at any given time.

One alternative I think would make sense is to lift the `SeenState` to a parent component so that any child Contents components can share a `SeenState`. These are the React docs for this technique (just so that we're on the same page of which technique I have in mind): https://react.dev/learn/sharing-state-between-components. Not 100% this is the best idea, but it would allow for grouping Contents components in the parent component.

Reply:
Nice, yea, to me this new approach is better because it's more clearly and robustly ensuring that glossary entries will be unique within an `Article`. That said, I am confused about how terms were being glossaried twice prior to this fix because it seems each Article already only had [a single Contents](https://github.com/StampyAI/stampy-ui/blob/9433f390c208c92cf71ee6cbbd84b936294025cc/app/components/Article/index.tsx#L167-L174). So the local `seen` tracking in `Contents` that was there before should have been equivalent.

What were the reproduction steps for this "double glossaried" issue prior to this fix?